### ISSUE TITLE
LInk Extras: context sub-menus 

### ISSUE THREAD

**Issue by markwr** [igorlogius/igorlogius#70]
### Describe your suggestion

Can I revert to the previous version, and if so, how?
Else, can you tell me how to achieve my goal* (below), using the current version?
Else if that is not possible, consider my suggestion to give an option in the next version so I can achieve my goal* (below).

GOAL: Be able to right click (after highlighting text) and see my preferred option right away in the top menu (without having to drill down into sub-menu choices).

History: In the prior version, I had it configured to only have one choice defined. Then that choice showed up right away in the right-click top menu.

**Comment by igorlogius** [comment]
You can use the "custom shortcuts" to invoke the action immediately
It's a bit unintuitive but that is partily the fault of the firefox developers. 

There are 10 free to assign "Custom Copy Actions" in the Shortcuts View for the addon
Each action corresponds to a column in the table in the Preferences 

![2024-12-12_01-53](https://github.com/user-attachments/assets/3d30b04d-01be-4e3f-bcc2-c2d005b3223e)


![2024-12-12_01-51](https://github.com/user-attachments/assets/c87c7a9e-7d11-45c5-a315-c34f50b62411)

So, just set a shortcut for any action you want to invoke and you dont even need to open a context menu anymore. 

Hope this is a sufficently efficient solution for your workflow.

**Comment by markwr** [comment]
Well, first of all, I'd like to say I'm impressed on how quickly you responded. I probably explained it a bit better in my review here:
https://addons.mozilla.org/en-US/firefox/addon/link-extras/reviews/
I'll be changing that review to 5 stars and reword it to mention your quick response to a question.
Below is a screen snip-it of how I had it configured before (and after) the update. That was the only choice added to my context menu (top-level). I am quite used to just tapping my right mouse button and tapping U to get my result. While a Shortcut/hotkey will help and is also quick, it is something new for me to remember (a slight challenge at the age of 74, lol).
For now, I'll define my action to Alt-U (which works) and hope it does not conflict with anything else in the future.
I still prefer to just move [URL Copy Link Extras - %text %url] to my top-level context menu if you know an easy way for me to that (I don't actually need any of the other sub-menus). If not, that's fine. I'll leave the review the way it is until tomorrow to give you time to read my description there. ps - I'm a retired database programmer (old school), but that's somewhat unrelated to your programming language/platform etc (I retired in 2006).
![Link Extras - Mark Wright](https://github.com/user-attachments/assets/b0ebd216-8936-47cd-bad8-8b9dd63c3d4d)

**Comment by igorlogius** [comment]
> not reverting to prior version

Without reverting or using a modified version that is a bit difficult since each addons can only register one context entry at the moment as far as i know.

 While i could add options to reduce the menu, it seems like something rather special ... so i think it would be best to either downgrade or use a modified version, rather than addon more options to modify the menu structure.

It wouldn't be much of an issue do create a custom stripped down version that only has the function you want, but i'll leave the decision up to you.

Just let me know and i'll share the modified xpi with you.

**Comment by markwr** [comment]
Thank you very much, I would be thrilled to try the modified version if you're still willing.

**Comment by markwr** [comment]
Thanks! I assume I should uninstall the current installed Link Extras extension before installing this one, correct?

**Comment by igorlogius** [comment]
>  I assume I should uninstall the current installed Link Extras extension before installing this one, correct?

yes, correct. 
You can still use the other options via the shortcuts, but the menu has been stripped down to only include the elements that are listed in the copy table. 

So by just setting only one Element as you had, everything should be as you want it.

**Comment by markwr** [comment]
Yahoo! It works perfectly! Thank you so much.
